_source/_posts/2023-01-23-keycloak-kubernetes-prod.md

```diff
@@ -0,0 +1,637 @@
+
+Click on **Add** to continue the configuration. Below the Client Secret field, click on **Advanced**. In the _Scopes_ field, set `openid profile email offline_access` and click on **Save**.
+
+On the left menu, choose **Authentication**. In the flows table, choose **browser**. In the _Identity Provider Redirector_ step, click the **gear** icon. Set an alias for the configuration, and set **auth0** as the default identity provider. This configuration will skip the Keycloak sign-in form, and display the Auth0 sign-in form directly.
+
+{% img blog/keycloak-kubernetes-prod/auth0-redirector.png alt:"Identity provider redirector to Auth0" width:"450" %}{: .center-image }
+
+Go back to **Authentication**, and choose the **first broker login** flow. Disable the _Review Profile_ step.
+
+### Map the roles claim
+
+The users that sign in through Auth0 are imported to Keycloak. The role assigned in Auth0 must be mapped to the pre-configured roles in Keycloak for the authorization mechanism to work. In the Keycloak Administration Console, go to **Identity providers** again, and choose **auth0**. Then choose the **Mappers** tab. Click **Add mapper**. Fill in the values as follows and click **Save**:
+
+- Name: **ROLE_USER**
+- Sync mode override: **Force** (update the user roles during every login)
+- Mapper type: **Claim to Role**
+- Claim: **https://www\\.jhipster\\.tech/roles** (escape the dot (.) with backslash (\\.))
+- Claim Value: **ROLE_USER**
+- Role: Select **ROLE_USER**
+
+Repeat the process for mapping `ROLE_ADMIN`.
+
+{% img blog/keycloak-kubernetes-prod/role-mapper.png alt:"Add mapper form" width:"800" %}{: .center-image }
+
+### Test Keycloak as an identity broker
+
+Navigate to **http://gateway.\<namespace\>.\<public-ip\>.nip.io**, the gateway home must display. Click on **sign in** and the browser should be redirected to the Auth0 sign-in form:
+
+{% img blog/keycloak-kubernetes-prod/auth0-login.png alt:"Add mapper form" width:"350" %}{: .center-image }
+
+After signing in, the gateway home page will display the username (email). If roles were mapped correctly, the user should be able to create entities (granted by ROLE_USER), and the Administration menu will also display (granted by ROLE_ADMIN).
+
+## Learn More about Keycloak in production
+
+I hope you enjoyed this post and learned about some best practices for deploying Keycloak to production when doing JHipster development. Keep learning about Keycloak and JHipster! Check out the following links:
+
+- [Integrate React Native and Spring Boot Securely](/blog/2022/10/12/integrate-react-native-and-spring-boot-securely)
+- [Use GitHub Actions to Build GraalVM Native Images](/blog/2022/04/22/github-actions-graalvm)
+- [Full Stack Java with React, Spring Boot, and JHipster](https://auth0.com/blog/full-stack-java-with-react-spring-boot-and-jhipster/)
+- [Introducing Spring Native for JHipster: Serverless Full-Stack Made Easy](/blog/2022/03/03/spring-native-jhipster)
```

Comment:
These are a bit old and we should update. Here's some recent ones:

- https://auth0.com/blog/developers/
- https://developer.okta.com/blog/tags/auth0/

Reply:
Should the linked posts be related (Keycloak, Spring Boot)?

Reply:
If possible, yes. I think it's OK to include JHipster too, or any Java-related post.

Reply:
Added newer posts.